src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/jpa/entity/cas1/Cas1SpaceSearchRepository.kt

```diff
@@ -0,0 +1,276 @@
+package uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.cas1
+
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate
+import org.springframework.stereotype.Repository
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.model.ApprovedPremisesType
+import java.sql.ResultSet
+import java.time.LocalDate
+import java.util.UUID
+
+private const val AP_TYPE_FILTER = """
+  AND result.ap_type IN (:apTypes)
+"""
+
+private const val GENDER_FILTER = """
+  AND EXISTS (
+    SELECT 1
+    FROM premises_characteristics pc
+    WHERE
+      pc.premises_id = result.premises_id
+      AND pc.characteristic_id IN (:genderCharacteristics)
+  )
+"""
+
+private const val PREMISES_CHARACTERISTICS_FILTER = """
+  AND EXISTS (
+    SELECT 1
+    FROM premises_characteristics pc
+    WHERE
+      pc.premises_id = result.premises_id
+      AND pc.characteristic_id IN (:premisesCharacteristics)
+  )
+"""
+
+private const val ROOM_CHARACTERISTICS_FILTER = """
+  AND EXISTS (
+    SELECT 1
+    FROM room_characteristics rc
+    JOIN rooms r
+    ON rc.room_id = r.id
+    WHERE
+      r.premises_id = result.premises_id
+      AND rc.characteristic_id IN (:roomCharacteristics)
+  )
+"""
+
+private const val CANDIDATE_PREMISES_QUERY_TEMPLATE = """
+SELECT
+  result.*
+FROM
+(
+  SELECT
+    p.id AS premises_id,
+    ST_Distance(
+      (SELECT point FROM postcode_districts pd WHERE pd.outcode = :outcode)::geography,
+      ap.point::geography
+    ) * 0.000621371 AS distance_in_miles,
```

Comment:
Can we have a constant for this number please, i'm not sure what it is?

Reply:
OK.I googled it : Conversion of Metre to Mile: Metres and miles are used to measure the distance or length in the metric system. A stretch of 0.000621371 metres makes a mile